sdk/servicebus/azure-servicebus/azure/servicebus/management/_models.py

```diff
@@ -25,8 +25,59 @@
 adjust_attribute_map()
 
 
-class QueueDescription(object):  # pylint:disable=too-many-instance-attributes
-    """Description of a Service Bus queue resource.
+class DictMixin(object):
+
+    def __setitem__(self, key, item):
+        self.__dict__[key] = item
+
+    def __getitem__(self, key):
+        return self.__dict__[key]
+
+    def __repr__(self):
+        return str(self)
+
+    def __len__(self):
+        return len(self.keys())
+
+    def __delitem__(self, key):
+        self.__dict__[key] = None
+
+    def __eq__(self, other):
+        """Compare objects by comparing all attributes."""
+        if isinstance(other, self.__class__):
+            return self.__dict__ == other.__dict__
+        return False
```

Comment:
so comparing by value not by reference is the pattern we applied in other sdks as well?

Reply:
Yes this is the same as in Storage Blob SDK